mycroft/audio/services/__init__.py

```diff
@@ -27,6 +27,26 @@ class AudioBackend(metaclass=ABCMeta):
     def __init__(self, config, bus):
         self._track_start_callback = None
         self.supports_mime_hints = False
+        self.bus = bus
+        self.config = config
+        self.index = 0
+        self.tracks = []
+        self.track_data = {}
+        self.bus.on('play:status', self.handle_track_status)
+
+    def handle_track_status(self, message):
+        index = message.data.get("playlist_position")
+        uri = message.data.get("uri")
+        if index is None and uri:
+            for idx in self.track_data:
+                if self.track_data[idx].get("uri", "") == uri:
+                    index = idx
+                    break
+        elif index is None:
+            index = len(self.track_data) - 1
```

Comment:
Can we trust the length of `self.track_data` here?

I assume a dict is used because it may receive data on any track without the preceding track data necessarily existing? 

So imagining:
- it receives only data about tracks with index's of 1 and 2
- then for some reason receives an index of None
- Would it therefore overwrite the track_data at index 1?

Reply:
I've cut this part of the PR since this was part of @JarbasAI WIP updates to the track handling in the audioservices.

This PR is now only the proposed standard methods for the the common play skills.

Reply:
I haven't had much time, but im still working on this. I think its better if thats a follow up PR :+1: